fili-core/src/test/groovy/com/yahoo/bard/webservice/async/AsyncInvalidApiRequest.groovy

```diff
@@ -0,0 +1,44 @@
+package com.yahoo.bard.webservice.async
+
+import com.yahoo.bard.webservice.util.GroovyTestUtils
+
+/**
+ * Verifies that when the user has a syntax error (or some other error that triggers an InvalidApiRequestException) that
+ * the error is returned immediately.
+ */
+class AsyncInvalidApiRequest extends AsyncFunctionalSpec {
+
+    String EXPECTED_ERROR_MESSAGE =
+            """{
+                    "status": 400,
+                    "statusName": "Bad Request",
+                    "reason": "com.yahoo.bard.webservice.web.BadApiRequestException",
+                    "description": "Required parameter metrics is missing or empty. Use 'metrics=METRICNAME1,METRICNAME2' in the query string.",
+                    "druidQuery": null
+            }"""
+
+    @Override
+    Map<String, Closure<String>> getResultsToTargetFunctions() {
+        [ data: { "data/shapes/day" } ]
+    }
+
+    @Override
+    Map<String, Closure<Void>> getResultAssertions() {
+        [ data: { assert GroovyTestUtils.compareJson(it.readEntity(String), EXPECTED_ERROR_MESSAGE) } ]
+    }
+
+    @Override
+    Map<String, Closure<Map<String, List<String>>>> getQueryParameters() {
+        return [
+                data: { [
+                        asyncAfter: ["0"],
+                        dateTime: ["2016-08-30/2016-08-31"]
+                ] }
+        ]
+    }
+
+    @Override
+    Closure<String> getFakeDruidResponse() {
+        return {"I will never leave this curly prison."}
```

Comment:
Rather than having to override a method when we don't care about the response, does it make sense to have the method be non-abstract with a default Druid response of an empty object, or list or something?